api/activity.go

```diff
@@ -67,44 +67,34 @@ const (
 
 func (e ActivityType) String() string {
 	switch e {
-	case ActivityIssueCreate:
-		return "bb.issue.create"
-	case ActivityIssueCommentCreate:
-		return "bb.issue.comment.create"
-	case ActivityIssueFieldUpdate:
-		return "bb.issue.field.update"
-	case ActivityIssueStatusUpdate:
-		return "bb.issue.status.update"
-	case ActivityPipelineTaskStatusUpdate:
-		return "bb.pipeline.task.status.update"
-	case ActivityPipelineTaskFileCommit:
-		return "bb.pipeline.task.file.commit"
-	case ActivityPipelineTaskStatementUpdate:
-		return "bb.pipeline.task.statement.update"
-	case ActivityMemberCreate:
-		return "bb.member.create"
-	case ActivityMemberRoleUpdate:
-		return "bb.member.role.update"
-	case ActivityMemberActivate:
-		return "bb.member.activate"
-	case ActivityMemberDeactivate:
-		return "bb.member.deactivate"
-	case ActivityProjectRepositoryPush:
-		return "bb.project.repository.push"
-	case ActivityProjectDatabaseTransfer:
-		return "bb.project.database.transfer"
-	case ActivityProjectMemberCreate:
-		return "bb.project.member.create"
-	case ActivityProjectMemberDelete:
-		return "bb.project.member.delete"
-	case ActivityProjectMemberRoleUpdate:
-		return "bb.project.member.role.update"
-	case ActivitySQLEditorQuery:
-		return "bb.sql-editor.query"
-	case ActivityDatabaseRecoveryPITRDone:
-		return "bb.database.recovery.pitr.done"
+	// Issue related.
+	case ActivityIssueCreate,
+		ActivityIssueCommentCreate,
+		ActivityIssueFieldUpdate,
+		ActivityIssueStatusUpdate,
+		ActivityPipelineTaskStatusUpdate,
+		ActivityPipelineTaskFileCommit,
+		ActivityPipelineTaskStatementUpdate,
+		ActivityPipelineTaskEarliestAllowedTimeUpdate,
+		// Member related
+		ActivityMemberCreate,
+		ActivityMemberRoleUpdate,
+		ActivityMemberActivate,
+		ActivityMemberDeactivate,
+		// Project related
+		ActivityProjectRepositoryPush,
+		ActivityProjectDatabaseTransfer,
+		ActivityProjectMemberCreate,
+		ActivityProjectMemberDelete,
+		ActivityProjectMemberRoleUpdate,
+		// SQL Editor related
+		ActivitySQLEditorQuery,
+		// Database related
+		ActivityDatabaseRecoveryPITRDone:
+		return string(e)
+	default:
+		return "bb.activity.unknown"
 	}
-	return "bb.activity.unknown"
```

Comment:
string(e) ?

Reply:
It's valid code to do `a := ActivityType("test")`, and `a.String()` will be `"test"`, which violates the activity table constraint check.

Reply:
Maybe we should prevent that to happen from beginning.